src/Audit.NET.AmazonQLDB/Providers/AmazonQldbDataProvider.cs

```diff
@@ -74,23 +75,36 @@ public AmazonQldbDataProvider(Action<IAmazonQldbProviderConfigurator> config)
         /// <summary>
         /// Inserts an event into AmazonQLDB
         /// </summary>
-        public override object InsertEvent(AuditEvent auditEvent)
+        public override object InsertEvent(AuditEvent auditEvent) => InsertEventAsync(auditEvent).GetAwaiter().GetResult();
+
+        /// <summary>
+        /// Asynchronously inserts an event into AmazonQLDB
+        /// </summary>
+        public override async Task<object> InsertEventAsync(AuditEvent auditEvent)
         {
             var driver = QldbDriver.Value;
             var tableName = GetTableName(auditEvent);
-
-            List<IIonValue> insertResult = driver.Execute(txn =>
+            var insertResult = new List<IIonValue>();
```

Comment:
I think this block could be cleaner if we call `ToListAsync()`:

```c#
var insertResult = await driver.Execute(async txn =>
{
    var json = auditEvent.ToJson();
    var insertInto = $@"INSERT INTO {tableName} VALUE ?";
    try
    {
        return await (await txn.Execute(insertInto, IonLoader.Default.Load(json))).ToListAsync();
    }
    catch (BadRequestException e) when (e.Message.Contains($"No such variable named '{tableName}'"))
    {
        await txn.Execute($"CREATE TABLE {tableName}");
        return await (await txn.Execute(insertInto, IonLoader.Default.Load(json))).ToListAsync();
    }
});
```

Reply:
Or even better, we can call the FirstAsync inside the lambda, and the entire method would be like:

```c#
public override async Task<object> InsertEventAsync(AuditEvent auditEvent)
{
    var driver = QldbDriver.Value;
    var tableName = GetTableName(auditEvent);
    var insertResult = await driver.Execute(async txn =>
    {
        var json = auditEvent.ToJson();
        var insertInto = $@"INSERT INTO {tableName} VALUE ?";
        try
        {
            return await (await txn.Execute(insertInto, IonLoader.Default.Load(json))).FirstAsync();
        }
        catch (BadRequestException e) when (e.Message.Contains($"No such variable named '{tableName}'"))
        {
            await txn.Execute($"CREATE TABLE {tableName}");
            return await (await txn.Execute(insertInto, IonLoader.Default.Load(json))).FirstAsync();
        }
    });

    var insertDocumentId = insertResult.GetField("documentId").StringValue;
    return (insertDocumentId, tableName);
}
```

Reply:
@thepirat000 thanks for checking this; I don't know why, but I had the impression there is no LINQ implementation for async iterators. I made the suggested updates.